File: swift/HeightChecker.swift
// https://leetcode.com/problems/height-checker/

class Solution {
    func heightChecker(_ heights: [Int]) -> Int {
        var count = 0
        let expected = heights.sorted { $0 < $1 }
        for i in stride(from: 0, to: heights.count, by: 1) {
            if heights[i] != expected[i] {
                count += 1
            }
        }
        return count;
    }
}

File: c/HeightChecker.c
// https://leetcode.com/problems/height-checker/


int heightChecker(int* heights, int heightsSize) {
    int expected[heightsSize];
    int x, c = 0;
    bool f =  false;
    
    for (int i=  0; i < heightsSize; i++)
        expected[i] = heights[i];
        
    for (int i = 0; i < heightsSize - 1; i++) 
        for (int k = 0; k < heightsSize - 1; k++)
            if (expected[k] >= expected[k + 1]) {
                x = expected[k];
                expected[k] = expected[k + 1];
                expected[k + 1] = x;
                f = true;
            }
    
    if (f == false)
        return 0;
    
    for (int i = 0; i < heightsSize; i++)
        if (expected[i] != heights[i])
            c++;
    
    return c;
}